src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -384,45 +446,51 @@ class JSONReader
         return false;
     }
 
-    boost::shared_ptr<const rapidjson::Document> _GetDocumentFromFilename(const std::string& fullFilename, rapidjson::Document::AllocatorType& alloc)
+    boost::shared_ptr<const rapidjson::Document> _GetDocumentFromFilename(const std::string& referenceUri, rapidjson::Document::AllocatorType& alloc)
```

Comment:
@JosephBerman Please explain the changes in this function.

Why do you need to add to `_urlsAlreadyStaged`? At this point, if you don't already have it in _rapidJSONDocuments, it is too late.

Why do you treat `msgpack` different from `json`?

Reply:
That was left over from when still using ifstream, removing.